### DISCUSSION TITLE
How to render markdoc content based on query parameters in nextJS

### DISCUSSION THREAD

**Discussion by aditya-ayare** [markdoc/markdoc#446] [Q&A]
I want to render some partials based on query parameters in one of the .md pages. Not sure how to achieve the dynamic behavior. I want to achieve something like : 
```
{% if includes($countries, "US") %}
{% partial file="us/placeholders.md" /%}
{% /if %}
{% if includes($countries, "IND") %}
{% partial file="ind/placeholders.md" /%}
{% /if %}
```

**Comment by mfix-stripe** [comment]
I would just read from the query params within your React components themselves. 

Which part from the example above is meant to come from the query params?

**Comment by aditya-ayare** [comment]
The scenario is, in my nextJS project I want to keep all the files in the pages folder in .md format. I want to read the URL query params in _app.tsx or any HOC (before the markdoc page is rendered) and change the markdoc variables based on query params and render page. In runtime. I'm not sure if it's possible to do it without having js in the pages folder.
